lib/Doctrine/Common/Collections/Collection.php

```diff
@@ -170,6 +209,8 @@ public function next();
      * @param Closure $p The predicate.
      *
      * @return bool TRUE if the predicate is TRUE for at least one element, FALSE otherwise.
+     *
+     * @psalm-param Closure(TKey, T):bool $p
```

Comment:
```suggestion
     * @psalm-param Closure(TKey=, T=):bool $p
```

Reply:
Why do you want the arguments to be optional? Is there a use case where you want to pass a closure in that doesn't work with either the value or the key?

Reply:
I can easily imagine passing a closure that works with keys only. But if you're able to ignore the values, why shouldn't you be able to omit keys as well? Granted, it's more useful for things like `filter` and `partition`, where you could want random sampling:
```php
$coll->filter(function() { return rand() < 0.1; /* take 10 % of the entries */ });
```
For `map`, you may want to construct the collection having the same keys, but some placeholder or random values:
```php
$item = new Entity;
$coll->map(function() use ($item) { return $item; });
```
`exists` and `forAll` are cases where ignoring both key and values is probably the least useful, so they included mostly for completeness, as it's technically not a problem passing no-args closures to them.

Reply:
Alright, yeah. Makes sense. Gonna add your suggestions 👍